snapcraft/__init__.py

```diff
@@ -225,6 +225,18 @@
     `prime/` directory reflects the file structure of the snap with no
     extraneous content).
 
+  - prepare: shell script
+
+    If present, the shell script defined here is run before the `build` step
+    of the plugin starts. The working directory is the base build
+    directory for the given part. The defined script is run with `/bin/sh`.
```

Comment:
I don't think we need to worry, all of snapcraft is tied to `dash` now and so are wrapper scripts and everything else